lib/jekyll/converters/smartypants.rb

```diff
@@ -29,7 +29,12 @@ def output_ext(_)
       end
 
       def convert(content)
-        Kramdown::Document.new(content, @config).to_html.chomp
+        document = Kramdown::Document.new(content, @config)
+        html_output = document.to_html.chomp
```

Comment:
Is there any reason why we're doing `chomp` here but not in `KramdownParser` above?

Reply:
My thinking was that SmartyPants only deals with inline elements (and not block elements), so white space is insignificant.